src/Orleans/Core/GrainClient.cs

```diff
@@ -229,7 +229,7 @@ private static void DoInternalInitialize(IInternalClusterClient clusterClient)
                         // before we make a call that makes RuntimeClient.Current not null
                         isFullyInitialized = false;
                         client = clusterClient;  // Keep reference, to avoid GC problems
-                        client.Start().Wait();
+                        client.Connect().Wait();
```

Comment:
sorry, for not bringing this up earlier, but I believe a good practice when we must do a blocking (either .Wait() or .Result) is to use `.GetAwaiter().GetResult()`, so that at least the exception is no longer an AggregateException, which would only be useful if you had more than 1 exception (for example if you call Task.WhenAll).
That's what I remember at least, and doing a quick search I get stuff like this: https://github.com/aspnet/Security/issues/59